src/dev/build/tasks/create_cdn_assets_task.ts

```diff
@@ -0,0 +1,98 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License
+ * 2.0 and the Server Side Public License, v 1; you may not use this file except
+ * in compliance with, at your election, the Elastic License 2.0 or the Server
+ * Side Public License, v 1.
+ */
+
+import { readFileSync } from 'fs';
+import { access } from 'fs/promises';
+
+import { resolve, dirname } from 'path';
+import { asyncForEach } from '@kbn/std';
+import { Jsonc } from '@kbn/repo-packages';
+
+import del from 'del';
+import globby from 'globby';
+
+import { mkdirp, compressTar, Task, copyAll } from '../lib';
+
+export const CreateCdnAssets: Task = {
+  description: 'Creating CDN assets',
+
+  async run(config, log, build) {
+    const buildSource = build.resolvePath();
+    const buildNum = config.getBuildNumber();
+    const buildVersion = config.getBuildVersion();
+    const assets = config.resolveFromRepo('build', 'cdn-assets');
+    const bundles = resolve(assets, String(buildNum), 'bundles');
+    const plugin = resolve(bundles, 'plugin');
+
+    await del(assets);
+    await mkdirp(assets);
+
+    // Plugins
+    await mkdirp(plugin);
+    const plugins = globby.sync([`${buildSource}/node_modules/@kbn/**/*/kibana.jsonc`]);
+    await asyncForEach(plugins, async (path) => {
+      const manifest = Jsonc.parse(readFileSync(path, 'utf8')) as any;
+      if (manifest?.plugin?.id) {
+        const pluginRoot = resolve(dirname(path));
+
+        try {
+          const assetsSource = resolve(pluginRoot, 'assets');
+          const assetsDest = resolve(plugin, manifest.plugin.id, 'assets');
+          await access(assetsSource);
+          await copyAll(assetsSource, assetsDest);
+        } catch (e) {
+          // assets are optional
+          if (!(e.code === 'ENOENT' && e.syscall === 'access')) throw e;
+        }
+
+        try {
+          const bundlesSource = resolve(pluginRoot, 'target', 'public');
+          const bundlesDest = resolve(plugin, manifest.plugin.id, '1.0.0');
```

Comment:
Is hard-coding `1.0.0` for the plugin's version really fine here? Did we remove the concept of plugin version with the new manifest format?

https://github.com/elastic/kibana/blob/1b8581540295fde746dae6b4a09d74fb5821bfef/packages/core/apps/core-apps-server-internal/src/bundle_routes/register_bundle_routes.ts#L78-L84

Reply:
It is hardcoded [1].  It sounds like plugin versions aren't relevant to us but useful to keep around for third party plugins [2]. 

1. [source](https://github.com/elastic/kibana/blame/7cd0b74ccb21e10407d2044b49fe0ff403e63451/packages/core/plugins/core-plugins-server-internal/src/discovery/plugin_manifest_from_plugin_package.ts#L2)
1. [discussion](https://github.com/elastic/kibana/pull/149370/files#r1085987275)